### DISCUSSION TITLE
v7 - weird useWatch issue

### DISCUSSION THREAD

**Discussion by terebentina** [react-hook-form/react-hook-form#4535] [General]
Hi,

Using v7 rc.7 I have this problem where useWatch doesn't show the latest value the first time after using setValue but it does every time afterwards.
Here's a csb showing the problem: https://codesandbox.io/s/setvalue-watch-484pv?file=/src/App.js

How to reproduce:
1. open the CSB above
2. note how the first line reads "preview:" when I would expect it to be "preview: modified"
3. note how the input has the correct value "modified"
4. type something into the input and the preview value changes and then stays in sync with what's in the input.

**Comment by bluebill1049** [comment]
`useWatch` will return `defaultValues` or `defaultValue` before `useForm` complete mounted. In your example, `useEffect` is running at the child component (first before useForm useEffect is ran) which triggered useWatch to return `defaultValues` instead of current form values.

**Reply by terebentina** [reply]
Hmm...I think I understand but I am not sure how I could go about "refreshing" useWatch then...any suggestion/best practices?
That is make it see the latest values without the user making any changes to the form.
